app/webpacker/components/Tickets/index.jsx

```diff
@@ -8,19 +10,91 @@ import TicketHeader from './TicketHeader';
 import TicketWorkbench from './TicketWorkbench';
 import TicketLogs from './TicketLogs';
 
+function SkateholderSelector({ stakeholderList, setUserSelectedStakeholder }) {
+  const [selectedOption, setSelectedOption] = useState(stakeholderList[0]);
+  return (
+    <>
+      <p>
+        You are part of more than one stakeholder, please select the stakeholder as which you want
+        to visit the ticket page.
+      </p>
+      <Dropdown
+        options={stakeholderList.map((requesterStakeholder) => ({
+          key: requesterStakeholder.id,
+          text: requesterStakeholder.stakeholder.name,
+          value: requesterStakeholder,
+        }))}
+        value={selectedOption}
+        onChange={(__, { value }) => setSelectedOption(value)}
+      />
+      <Button
+        disabled={!selectedOption}
+        onClick={() => setUserSelectedStakeholder(selectedOption)}
+      >
+        Select
+      </Button>
+    </>
+  );
+}
+
+function TicketContent({ ticketDetails, currentStakeholder, sync }) {
+  return (
+    <>
+      <TicketHeader ticketDetails={ticketDetails} />
+      <TicketWorkbench
+        ticketDetails={ticketDetails}
+        sync={sync}
+        currentStakeholder={currentStakeholder}
+      />
+      <TicketLogs logs={ticketDetails.ticket.ticket_logs} />
+    </>
+  );
+}
+
 export default function Tickets({ id }) {
   const {
     data: ticketDetails, sync, loading, error,
   } = useLoadedData(actionUrls.tickets.show(id));
 
+  const [userSelectedStakeholder, setUserSelectedStakeholder] = useState();
+  const currentStakeholder = userSelectedStakeholder || ticketDetails?.requester_stakeholders[0];
+  const shouldUserSelectStakeholder = (
+    !userSelectedStakeholder
+    && ticketDetails?.requester_stakeholders?.length > 1);
+
   if (loading) return <Loading />;
   if (error) return <Errored />;
 
   return (
-    <Container fluid>
-      <TicketHeader ticketDetails={ticketDetails} />
-      <TicketWorkbench ticketDetails={ticketDetails} sync={sync} />
-      <TicketLogs logs={ticketDetails.ticket.ticket_logs} />
-    </Container>
+    <>
+      <Container fluid>
+        {userSelectedStakeholder && (
+          <Message>
+            {`You are currently viewing the ticket as stakeholder "${userSelectedStakeholder.stakeholder.name}".`}
+            <Button
+              onClick={() => setUserSelectedStakeholder(false)}
+            >
+              Click here to change
+            </Button>
+          </Message>
+        )}
+        {!shouldUserSelectStakeholder && (
+          <TicketContent
+            ticketDetails={ticketDetails}
+            currentStakeholder={currentStakeholder}
+            sync={sync}
+          />
+        )}
+      </Container>
+      <Modal open={shouldUserSelectStakeholder}>
```

Comment:
It seems that you "force" this modal open whenever the user has to make a choice. This is good, but have you made sure that the default SemUI behavior prevents the user from closing the modal by clicking outside the modal window frame?

I know it's definintely possible to set this "forced modal" mode in SemUI, I just don't know whether it's the default or not.

Reply:
I did a bit of research. I initially thought that by default it was forced modal as I couldn't close the modal by clicking outside or pressing escape button.

But after research I realized that it was acting as forced modal because I didn't had `onClose` in the modal. When I printed logs in `onClose`, it got printed when I click outside or press escape. So, I made `closeOnEscape` and `closeOnDimmerClick` to `false`. I didn't write `onClose` because I don't have to do any action there.